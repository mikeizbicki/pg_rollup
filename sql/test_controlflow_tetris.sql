SET client_min_messages TO WARNING;
create or replace language plpython3u;
create extension if not exists pg_rollup;

create table testtetris (
    id serial primary key,
    num int,
    val boolean
);

CREATE MATERIALIZED VIEW testtetris_rollup1 AS (
    SELECT
        sum(num),
        bool_and(val)
    FROM testtetris
);

select pgrollup_manage_all();

select * from testtetris_rollup1_raw;

-- FIXME: add more column types

drop table testtetris cascade;
