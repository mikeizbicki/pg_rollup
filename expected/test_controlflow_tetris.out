SET client_min_messages TO WARNING;
create or replace language plpython3u;
create extension if not exists pg_rollup;
create temporary table testtetris (
    id serial primary key,
    num int,
    val boolean
);
select pgrollup($$
CREATE INCREMENTAL MATERIALIZED VIEW testtetris_rollup1 AS (
    SELECT
        sum(num),
        bool_and(val)
    FROM testtetris
);
$$);
 pgrollup 
----------
 
(1 row)

select * from testtetris_rollup1_raw;
 sum(num) | bool_and(val) | raw_true 
----------+---------------+----------
        0 | t             | t
(1 row)

-- FIXME: add more column types
